src/Common/Auth/OpenIDConnect/Repositories/ClientRepository.php

```diff
@@ -47,7 +38,27 @@ public function getClientEntity($clientIdentifier, $grantType = null, $clientSec
 
     public function validateClient($clientIdentifier, $clientSecret, $grantType): bool
     {
-        // password and refresh grant
-        return true;
+        if ($grantType == 'authorization_code') {
+            $client = sqlQueryNoLog("SELECT `client_secret`, `is_confidential` FROM `oauth_clients` WHERE `client_id` = ?", [$clientIdentifier]);
+
```

Comment:
good catch. I knew about and put the method in then forgot about it. I started with a lesser version before putting in OIDC layer. Glad you caught because we weren't val secret.